### DISCUSSION TITLE
NXP MCXN947 GPIO5 support missing

### DISCUSSION THREAD

**Discussion by ECleverito** [zephyrproject-rtos/zephyr#81554] [General] [answered]
When attempting to build for MCXN947 with the Gpio5 module enabled in my overlay as such:

```
&gpio5 {
	status = "okay";
};
```

The following compile time error is generated for a missing 'kCLOCK_Gpio5' enum definition:

```
/opt/zephyrproject/zephyr/boards/nxp/frdm_mcxn947/board.c: In function 'frdm_mcxn947_init':
/opt/zephyrproject/zephyr/boards/nxp/frdm_mcxn947/board.c:204:27: error: 'kCLOCK_Gpio5' undeclared (first use in this function); did you mean 'kCLOCK_Gpio4'?
  204 |         CLOCK_EnableClock(kCLOCK_Gpio5);
      |                           ^~~~~~~~~~~~
      |                           kCLOCK_Gpio4
/opt/zephyrproject/zephyr/boards/nxp/frdm_mcxn947/board.c:204:27: note: each undeclared identifier is reported only once for each function it appears in
```

This occurs in the frdm_mcxn947's board.c initialization function:
```
#if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(gpio5))
	CLOCK_EnableClock(kCLOCK_Gpio5);
#endif
```

By manually creating entries for kCLOCK_Gpio5 in the GPIO_CLOCKS macro and the _clock_ip_name enum, I am able to fix this error:

hal_nxp/mcux/mcux-sdk/devices/MCXN947/drivers/fsl_clock.h @ line 86:
```
#define GPIO_CLOCKS                                                                        \
    {                                                                                      \
        kCLOCK_Gpio0, kCLOCK_Gpio1, kCLOCK_Gpio2, kCLOCK_Gpio3, kCLOCK_Gpio4, kCLOCK_Gpio5 \
    }
/*! @brief Clock ip name array for PINT. */
```

hal_nxp/mcux/mcux-sdk/devices/MCXN947/drivers/fsl_clock.h @ line 358:
```
    kCLOCK_Gpio4     = CLK_GATE_DEFINE(AHB_CLK_CTRL0, 23),   /*!< Clock gate name: Gpio4. */
    kCLOCK_Gpio5     = CLK_GATE_DEFINE(AHB_CLK_CTRL0, 24),   /*!< Clock gate name: Gpio5. */
    kCLOCK_Pint      = CLK_GATE_DEFINE(AHB_CLK_CTRL0, 25),   /*!< Clock gate name: Pint. */
```

It would make sense to create an issue instead in the HAL_NXP repo but it is not currently open for issues publicly.

**Comment by ECleverito** [comment]
@danieldegrasse @dleach02 @mmahadevan108 @DerekSnell

**Comment by DerekSnell** [comment]
Hi @ECleverito ,

Those GPIO clock gate enumerations are related to the bits in the register AHBCLKCTRL0, shown below.  Notice that GPIO5 is not included.  GPIO5 is special in this MCXNx4x device because it is in the VBAT domain, and is available in all power modes.

But since there is no bit for GPIO5 in this register, there is no enum for it in the HAL header file.
![image](https://github.com/user-attachments/assets/2596560b-84ac-461f-93f1-6aad10f4ce04)

Best regards

**Reply by ECleverito** [reply]
Ok @DerekSnell, understood. I noticed GPIO5 was working regardless of these commands, so that lines up.


Should probably remove the #ifdef from the frdm_mcxn947 board.c file then, no?

**Reply by mmahadevan108** [reply, marked as answer]
PR to address this. https://github.com/zephyrproject-rtos/zephyr/pull/81730.
Thank you for catching this.
